src/value-parser.js

```diff
@@ -335,7 +335,8 @@ function valueParse(parser, metaData, options, callback) {
                     valueMetaData.collation = collation;
                     if (baseType === 0xA5 || baseType === 0xAD || baseType === 0xA7 || baseType === 0xAF || baseType === 0xE7 || baseType === 0xEF) {
                       return readDataLength(parser, valueMetaData.type, {}, null, (maxDataLength) => {
-                        valueMetaData.dataLength = maxDataLength;
+                        // skip the 2-byte max length sent for BIGVARCHRTYPE, BIGCHARTYPE, NVARCHARTYPE, NCHARTYPE, BIGVARBINTYPE and BIGBINARYTYPE types
+                        // and parse based on the length of actual data
```

Comment:
I did look into '2.2.5.5.4 SQL_VARIANT Values' while investigating. It isn't clear why `max length` is passed and what it should be used for, as it is not the actual data length in TDS packet. 

Say for this query that returns variant, `select SERVERPROPERTY('Edition') as Version` , in TDS stream the max length portion of `VARIANT_PROPERTIES` is `8000`(wtr varchar), so there is a wait for all `8000` bytes to arrive, whereas the actual data has already arrived. This wait might also  consume data of different column or row(to fill 8k buffer). So, it is safe to read based on actual length.

I stepped into SqlClient to double check, and noticed those bytes were skipped with the comment 'skip over unknown properties'.
I did ad hoc testing with multiple rows and columns of variant and non-variant data mixture, dint run into any issue. Let me know if there is any scenario that you think might cause problem or needs to be tested.

P.S: Is the Travis failure intermittent?